tensorflow/contrib/rnn/python/ops/rnn_cell.py

```diff
@@ -1987,28 +1996,34 @@ def state_size(self):
   def output_size(self):
     return self._output_size
 
-  def _get_input_for_group(self, inpt, group_id, group_size):
+  def _get_input_for_group(self, inputs, group_id, group_size):
     """Slices inputs into groups to prepare for processing by cell's groups
 
     Args:
-      inpt: inputs
-      group_id: group id, for which to prepare extract input_group_id
+      inputs: cell input or it's previous state,
+              a Tensor, 2D, [batch x num_units]
+      group_id: group id, a Scalar, for which to prepare input
       group_size: size of the group
 
     Returns:
-      subset of inputs corresponding to group "group_id"
+      subset of inputs corresponding to group "group_id",
+      a Tensor, 2D, [batch x num_units/number_of_groups]
     """
-    return array_ops.slice(input_=inpt,
+    batch_size = inputs.get_shape()[0].value
+    if batch_size is None:
+      raise ValueError("Could not infer batch size from input")
```

Comment:
actually i think you may be able to just use `size=[batch_size, group_size]` even if batch_size is a tensor.  see if that works.